### ISSUE TITLE
oslogin: compiling error

### ISSUE THREAD

**Issue by helen-fornazier** [GoogleCloudPlatform/compute-image-packages#650]
I am trying to compile oslogin in a centos7 machine, and I get the error below.
I can fix the issue by reverting commit "Add user name validation to pam modules" #633
I'll take a deeper look, just adding here to track it.

```
[root@koike-centos7 google_compute_engine_oslogin]# make
g++ -fPIC -I/usr/include/json-c -c utils/oslogin_utils.cc -o utils/oslogin_utils.o
In file included from /usr/include/c++/4.8.2/regex:35:0,
                 from utils/oslogin_utils.cc:25:
/usr/include/c++/4.8.2/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support for the \
  ^
utils/oslogin_utils.cc: In function ‘bool oslogin_utils::ValidateUserName(const string&)’:
utils/oslogin_utils.cc:230:3: error: ‘regex’ is not a member of ‘std’
   std::regex r(kUserNameRegex);
   ^
utils/oslogin_utils.cc:230:14: error: expected ‘;’ before ‘r’
   std::regex r(kUserNameRegex);
              ^
utils/oslogin_utils.cc:231:10: error: ‘regex_match’ is not a member of ‘std’
   return std::regex_match(user_name, r);
          ^
utils/oslogin_utils.cc:231:38: error: ‘r’ was not declared in this scope
   return std::regex_match(user_name, r);
                                      ^
make: *** [utils/oslogin_utils.o] Error 1
```

**Comment by helen-fornazier** [comment]
regex is not supported in gcc older then 4.9
https://stackoverflow.com/a/8061172/2075374

**Comment by rjschwei** [comment]
The "fix" provided in b107105 is incomplete.

In versions prior to gcc 4.9 the code now depends on boost via

`#include <boost/regex.hpp>`

thus the Makefile needs to be updated to link against boost_regex, i.e. '-lboost_regex' when appropriate. Also there should be a note of this somewhere that boost is a dependency when built with gcc < 4.9
